File: RTPWindow.py
#keeps track of the window pointers, and whether packets are received or not, etc
class RTPWindow:
	def __init__(self,mini,maxi,numPackets):
		self.mini = mini
		self.maxi = maxi
		#A list of booleans with index referring to pointer of window and T/F if packet received
		self.rList = []
		for i in range(numPackets):
			self.rList.append(False)

	def setMax(self,val):
		self.maxi = val

	def getMax(self):
		return self.maxi

	def getMin(self):
		return self.mini

	def setMin(self,val):
		self.mini = val

	def getList(self):
		return self.rList

	def setList(self,pos,val):
		self.rList[pos] = val
		

File: RTPWindow.java
public class RTPWindow {
	private int min;
	private int max;
	private int numPackets;
	private boolean[] rList;

	public RTPWindow(int min, int max, int numPackets) {
		this.min = min;
		this.max = max;
		this.numPackets = numPackets;
		this.rList = rList;

		rList = new boolean[numPackets];
		for(int i = 0; i < numPackets; i++) {
			rList[i] = false;
		}
	}

	public int getMin() {
		return min;
	}

	public int getMax() {
		return max;
	}

	public boolean[] getrList() {
		return rList;
	}

	public void setMin(int val) {
		min = val;
	}

	public void setMax(int val) {
		max = val;
	}

	public void setrList(int pos, boolean val) {
		rList[pos] = val;
	}
}